source/common/http/alternate_protocols_cache_impl.cc

```diff
@@ -204,13 +226,39 @@ AlternateProtocolsCacheImpl::findAlternatives(const Origin& origin) {
   }
   if (key_value_store_ && original_size != protocols.size()) {
     key_value_store_->addOrUpdate(originToString(origin),
-                                  originDataToStringForCache(protocols, entry_it->second.srtt));
+                                  originDataToStringForCache(entry_it->second));
   }
 
   return makeOptRef(const_cast<const std::vector<AlternateProtocol>&>(protocols));
 }
 
 size_t AlternateProtocolsCacheImpl::size() const { return protocols_.size(); }
 
+Http3StatusTrackerSharedPtr
+AlternateProtocolsCacheImpl::getHttp3StatusTracker(const Origin& origin) {
+  auto entry_it = protocols_.find(origin);
+  if (entry_it == protocols_.end()) {
+    return nullptr;
+  }
+  if (entry_it->second.h3_status_tracker == nullptr) {
+    entry_it->second.h3_status_tracker =
+        std::make_shared<Http3StatusTrackerImpl>(dispatcher_, origin, *this);
+  }
+  return entry_it->second.h3_status_tracker;
+}
+
+void AlternateProtocolsCacheImpl::onHttp3StatusChanged(
```

Comment:
If we chose not to persist the http/3 status in the key value store then we can get rid of this callback (I believe) and all the associated plumbing.